packages/components/text-field/src/text-field.js

```diff
@@ -0,0 +1,163 @@
+			id: { type: String, reflect: true },
+			/** Value of the text field */
+			value: { type: String, reflect: true },
+			/** Pass type of the input element, if it's not multiline */
+			type: { type: String, reflect: true },
+			/** Placeholder of the text field */
+			placeholder: { type: String, reflect: true },
+			/** Array of messages to pass to helpt-text component. See help-text component for more info  */
+			helpTextMessages: { type: Array, reflect: true, attribute: 'help-text-messages' },
+			/** If you have more than one help text message , you should pass a title to it. See help-text component for more info  */
+			helpTextTitle: { type: String, reflect: true, attribute: 'help-text-title' },
+			/** Error messages to show underneath of the input when it has error */
+			errorMessages: { type: Array, reflect: true, attribute: 'error-messages' },
+			/** Error title, if there are more than one error message */
+			errorTitle: { type: Array, reflect: true, attribute: 'error-title' },
+			/** If the text field has an error, to show error messages and change the style of the input */
+			hasError: { type: Boolean, reflect: true, attribute: 'has-error' },
+			/** To show the astrix in the label, not doing validation yet */
+			required: { type: Boolean, reflect: true },
+			/** Is the text field disabled? */
+			disabled: { type: Boolean, reflect: true },
+			/** Is the text field readonly? */
+			readonly: { type: Boolean, reflect: true },
+			/** Will show a textarea instead of an input */
+			multiline: { type: Boolean, reflect: true },
+			/** Direction 'rtl' or 'ltr' */
+			dir: { type: String, reflect: true }
+		};
+	}
+
+	get direction() {
+		return this.dir ? this.dir : this.bodyDir;
+	}
+
+	onInput(event) {
+		/**
+		 * Emitted onInput event of input/textarea
+		 * @payload { value, originalEvent }
+		 */
+		this.dispatchCustomEvent('input', {
+			value: event.target.value,
+			id: this.id,
+			originalEvent: event
+		});
+	}
+
+	onChange(event) {
+		/**
+		 * Emitted onChange event of input/textarea
+		 * @payload { value, id, originalEvent }
+		 */
+		this.dispatchCustomEvent('change', {
+			value: event.target.value,
+			id: this.id,
+			originalEvent: event
+		});
+	}
+
+	get labelHtml() {
+		return html`
+			<label for=${this.id}>
+				<!-- If you want to have custom html in label, you can use this slot  -->
+				<slot name="label"><span>${this.label}</span></slot>
+				${this.required ? html` <span title="Required" class="required-astrix">*</span> ` : ''}
+			</label>
+		`;
+	}
+
+	get helpText() {
+		if (!this.helpTextMessages) {
+			return;
+		}
+
+		const messages = this.hasError ? this.errorMessages : this.helpTextMessages;
+		const type = this.hasError ? 'error' : undefined;
+		if (messages.length > 1) {
+			const title = this.hasError ? this.errorTitle : this.helpTextTitle;
+			return html`
+				<ts-help-text
+					type=${type}
+					title=${title}
+					messages=${JSON.stringify(messages)}
```

Comment:
You can use a property instead of attribute and do not stringify value. e.g.:
```
.messages=${messages}
```

Reply:
Ahh, you are right! finally a usecase for this `.` properties 😅